### PR TITLE
Add Haplocheck

### PR DESCRIPTION
**Pull request by lmtani**
## PR checklist

Closes https://github.com/nf-core/raredisease/issues/111

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/haplocheck/meta.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  - txt:
+      type: file
+      description: Raw report in txt format
+      pattern: "*.{txt}"
+
+  - html:
+      type: file
+      description: Haplocheck HTML report
+      pattern: "*.{txt}"
```
Comment (jemten) [inline modules/haplocheck/meta.yml:52]:
```suggestion
      pattern: "*.{html}"
```

File: modules/haplocheck/main.nf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::haplocheck=1.3.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/haplocheck:1.3.3--h4a94de4_0':
+        'quay.io/biocontainers/haplocheck:1.3.3--h4a94de4_0' }"
+
+    input:
+
```
Comment (jemten) [inline modules/haplocheck/main.nf:11]:
```suggestion
```

File: tests/modules/haplocheck/main.nf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+nextflow.enable.dsl = 2
+
+include { HAPLOCHECK } from '../../../modules/haplocheck/main.nf'
+
+workflow test_haplocheck {
+
+    input = [
+        [ id:'test' ], // meta map
+        // TODO: change to "params.test_data[]"
```
Comment (jemten) [inline tests/modules/haplocheck/main.nf:11]:
```suggestion
```

File: tests/modules/haplocheck/test.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml haplocheck
```
Comment (jemten) [inline tests/modules/haplocheck/test.yml:2]:
```suggestion
```

**Review by jemten — COMMENTED** [review]
Some quick comments @lmtani

**Comment by Lucpen** [comment]
@lmtani it would be good if you added stubs so that the tests are faster. You can find an example on how to do so on samtools view

**Comment by lmtani** [comment]
@jemten and @Lucpen : could you please check it again?

**Review by Lucpen — APPROVED** [review]
LGTM

**Pull request merged**